### PR TITLE
[Bookmarks] Add `areAllCategoriesEmpty` method to the bookmarks manager

### PR DESCRIPTION
**Pull request by kirylkaveryn**
This method is needed to implement the following features:
- hide the `export all` button (https://github.com/organicmaps/organicmaps/issues/8055)
- disable the `backup` button into the icloud sync enabling alert

Also, I've made a small refactoring of the `isCategoryNotEmpty` method to use the already `IsCategoryEmpty` which checks the `empty` (rather than the `cont`).

### CODE REVIEW

File: iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.h [hunk 1]
```diff
@@ -37,6 +37,7 @@ NS_SWIFT_NAME(BookmarksManager)
 - (BOOL)areBookmarksLoaded;
 - (void)loadBookmarks;
 
+- (BOOL)isAllCategoriesEmpty;
```
Comment (biodranik) [inline iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.h:40]: areAllCategoriesEmpty

File: iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm [hunk 2]
```diff
@@ -194,9 +194,19 @@ - (void)loadBookmarks
...
+{
+  auto const & list = self.bm.GetUnsortedBmGroupsIdList();
+  for (auto const & groupId : list) {
+    if (!self.bm.IsCategoryEmpty(groupId))
+      return NO;
+  }
+  return YES;
+}
+
+- (BOOL)isCategoryNotEmpty:(MWMMarkGroupID)groupId
```
Comment (biodranik) [inline iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm:207]: isCategoryEmpty

using Not complicates the understanding.

File: iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm [hunk 3]
```diff
@@ -194,9 +194,18 @@ - (void)loadBookmarks
 
 #pragma mark - Categories
 
-- (BOOL)isCategoryNotEmpty:(MWMMarkGroupID)groupId {
-  return self.bm.HasBmCategory(groupId) &&
-  (self.bm.GetUserMarkIds(groupId).size() + self.bm.GetTrackIds(groupId).size());
+- (BOOL)areAllCategoriesEmpty
```
Comment (biodranik) [inline iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm:197]: Would it be better/faster/more reusable on Android if implemented in C++?

**Review by biodranik — APPROVED** [review]

File: iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm (hunk 3 above)
Comment (kirylkaveryn) [reply iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm:197]: I think yes! I can try to move this logic into the core.

Comment (kirylkaveryn) [reply iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm:197]: I've moved this logic into the cpp.

**Review by biodranik — APPROVED** [review]
Cool, thanks!

**Pull request merged**